src/firefly/java/edu/caltech/ipac/firefly/server/query/lsst/LSSTCataLogSearch.java

```diff
@@ -181,18 +190,88 @@ private DataGroup getTableDataFromJson(File jsonFile, TableDef tableDef) throws
 
         return dg;
     }
-   /* private TableDef getMeta(TableServerRequest request){
+    /**
+     * This method translates the mySql data type to corresponding java data type
+     * @param classType
+     * @return
+     */
+    private Class getDataClass(String classType){
+
+        if (classType.equalsIgnoreCase("double")){
+            return Double.class;
+        }
+        else if (classType.equalsIgnoreCase("float") || classType.equalsIgnoreCase("real") ){
+            return Float.class;
+        }
+        else if (classType.equalsIgnoreCase("int(11)")){
+            return Integer.class;
+        }
+        else if (classType.equalsIgnoreCase("BigInt(20)") ){
+            return Long.class;
+        }
+        else if (classType.equalsIgnoreCase("bit(1)")){
+            return Boolean.class;
+        }
+        else if (classType.equalsIgnoreCase("TINYINT")){
+            return Byte.class;
+        }
+        else if (classType.equalsIgnoreCase("SMALLINT")){
+            return Short.class;
+        }
+        else if (classType.equalsIgnoreCase("string") ) {
+
+            return String.class;
+
+        }
+        else {
+            System.out.println(classType + "is not supported");
+        }
+        return null;
+
+    }
+    private  DataType[] getMetaFromMetatable(DataGroup dataGroup){
+
+        DataType[] dataTypes = new DataType[dataGroup.size()];
+        DataObject[] dataObjects=dataGroup.values().toArray(new DataObject[0]);
+        for (int i=0; i<dataObjects.length; i++){
+            dataTypes[i] = new DataType((String) dataObjects[i].getDataElement("Field"),
+                    getDataClass( (String) dataObjects[i].getDataElement("Type") ) );
+            boolean maybeNull = dataObjects[i].getDataElement("Null").toString().equalsIgnoreCase("yes")?true:false;
+            dataTypes[i] = new DataType( (String) dataObjects[i].getDataElement("Field"),
+                    (String) dataObjects[i].getDataElement("description"),
```

Comment:
You are using "description" as the column's label.  DataType also has a shortDescription(one sentence) field used to describe the column.  Verify that "description" here is the label and not the description of the column.